### PR TITLE
tests: Test pac-ret flag merging on clang with LTO

### PR DESCRIPTION
**Pull request by mrkajetanp**
Extend the test for pac-ret with clang and LTO by checking that different branch protection flags are preserved after the LTO step. There was an issue in older LLVM versions that was causing this to behave incorrectly.

try-job: aarch64-gnu-debug

### CODE REVIEW

File: tests/run-make/pointer-auth-link-with-c-lto-clang/rmake.rs [hunk 1]
```diff
@@ -32,5 +35,15 @@ fn main() {
         .input("test.rs")
         .output("test.bin")
         .run();
+
+    // Check that both a-key and b-key pac-ret survived LTO
```
Comment (jieyouxu) [inline tests/run-make/pointer-auth-link-with-c-lto-clang/rmake.rs:39]: Question: is there any background / links for this? I'm fine with the change, but it's even better if there's any kind of context to read.

Comment (mrkajetanp) [reply tests/run-make/pointer-auth-link-with-c-lto-clang/rmake.rs:39]: For sure, this is the context:
https://github.com/llvm/llvm-project/commit/1782810b8440144a0141c24192acbaeb55a1545d

TLDR: there was a bug in LLVM where during the LTO step instructions like "pacibsp" would get replaced with "paciasp" due to LLVM module flag merging behaviour. As currently written the assembly output should have no PAuth on library functions, `paciasp` at the start of main and `pacibsp` at the start of `foo`.
 
It is worrying that the test is failing, I'll try to reproduce it locally.

Comment (jieyouxu) [reply tests/run-make/pointer-auth-link-with-c-lto-clang/rmake.rs:39]: I agree, the test failing is... suspicious lol.

Comment (jieyouxu) [reply tests/run-make/pointer-auth-link-with-c-lto-clang/rmake.rs:39]: Also it might help if you do something like `let clang_version = clang().arg("-v").run();` and then print that to double-check what clang version CI has.

Comment (mrkajetanp) [reply tests/run-make/pointer-auth-link-with-c-lto-clang/rmake.rs:39]: Okay this is very funny
```
00000000000677a4 <test::main>:
   677a4: d503233f      paciasp
   677a8: d50323bf      autiasp
   677ac: d65f03c0      ret
```
The test fails because the compiler optimises away the entire call to `foo` and its pac instructions along with it. I'll fix it..

Comment (jieyouxu) [reply tests/run-make/pointer-auth-link-with-c-lto-clang/rmake.rs:39]: Right... That can explain it lol. `black_box` to the rescue.

**Comment by mrkajetanp** [comment]
Dropping that opt flag makes it work for me, could you have bors try it again?

**Review by jieyouxu — APPROVED** [review]
So it turns out not optimizing out the function that we want to check makes it easier to test for assembly inside said function 😆. Thanks!
